ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpImplTreeBuilder.fs

```diff
@@ -169,11 +167,18 @@ type FSharpImplTreeBuilder(lexer, document, decls, lifetime, projectedOffset, li
                 | TyconStruct -> ElementType.STRUCT_DECLARATION
                 | _ -> ElementType.OBJECT_TYPE_DECLARATION
 
-        for m in members do
-            x.ProcessTypeMember m
+        x.ProcessTypeMembers(members)
         x.EnsureMembersAreFinished()
         x.Done(range, mark, elementType)
 
+    member x.ProcessTypeMembers(typeMembers: SynMemberDefns) =
+        if typeMembers.Length = 0 then () else
```

Comment:
There's no need to calculate the whole list length if you're only interested in checking if the list is empty.